mods/qts/init.lua

```diff
@@ -15,8 +15,22 @@ dofile(qts.path.."/customversions.lua")
 --most players won't need this to work, and most should leave it off. Its for GIT integration into version watermarking.
 qts_internal.insecure = minetest.request_insecure_environment()
 dofile(qts.path.."/watermark.lua")
+local insecure = qts_internal.insecure
 qts_internal.insecure = nil
 
+--load FFI
+if insecure == nil then
+    minetest.log("error", "qts requires an insecure environment. Please add it to the list of allowed insecure environments.")

```

Comment:
Could we have it display exactly why it requires this?

Reply:
Updated in latest.